### PR TITLE
doc: nrf_security name change

### PR DESCRIPTION
**Pull request by mia-ko**
Changing the name of Nordic Security Module
to remove the module reference
(as it is a subsystem and not actually a module)

### CODE REVIEW

File: doc/nrf/libraries/nrf_security/doc/drivers.rst [hunk 1]
```diff
@@ -16,7 +16,7 @@ Enabling more than one PSA driver might add support for additional key sizes or
 
 It is possible to disable specific features on the PSA driver level to optimize the code size.
 
-The nrf_security module supports the following PSA drivers:
+The nrf_security supports the following PSA drivers:
```
Comment (joerchan) [inline doc/nrf/libraries/nrf_security/doc/drivers.rst:19]:
```suggestion
The nRF Security subsystem supports the following PSA drivers:
```

File: doc/nrf/libraries/nrf_security/doc/drivers.rst [hunk 2]
```diff
@@ -7,7 +7,7 @@ nrf_security drivers
    :local:
    :depth: 2
 
-The nrf_security module supports multiple enabled PSA drivers at the same time.
+The nrf_security supports multiple enabled PSA drivers at the same time.
```
Comment (joerchan) [inline doc/nrf/libraries/nrf_security/doc/drivers.rst:10]:
```suggestion
The nRF Security subsystem supports multiple enabled PSA drivers at the same time.
```

File: subsys/nrf_security/Kconfig [hunk 3]
```diff
@@ -32,13 +32,13 @@ config NORDIC_SECURITY_BACKEND
 
 config NRF_SECURITY
 	bool
-	prompt "Use nrf_security module" if !PSA_PROMPTLESS
+	prompt "Use nrf_security" if !PSA_PROMPTLESS
```
Comment (joerchan) [inline subsys/nrf_security/Kconfig:35]: I'm wondering if we should change the wording to "Enable" here.

```suggestion
	prompt "Enable nRF Security" if !PSA_PROMPTLESS
```

**Review by joerchan — CHANGES_REQUESTED** [review]
I don't see the need to use "nrf_security" in the documentation, I suggest we get rid of it.

**Review by mswarowsky — APPROVED** [review]
With the changes suggested by @joerchan lgtm

File: subsys/nrf_security/Kconfig (hunk 3 above)
Comment (mia-ko) [reply subsys/nrf_security/Kconfig:35]: Sure, it's more in line with other config options as well.

**Review by joerchan — APPROVED** [review]

**Review by melwee01 — APPROVED** [review]

**Pull request merged**